### ISSUE TITLE
v2.1 client not granting signature against workflow v2.0.0

### ISSUE THREAD

**Issue by jackfrancis** [deis/workflow-cli#122]
Using the newest version of the CLI (`2.1`) I am unable to login to older versions of the workflow platform (tested on `v2.0.0`). Specifically, no signature is being retained by the client. E.g.:

```
$ deis login <a v2.0.0 workflow endpoint>
!    WARNING: Client and server API versions do not match. Please consider upgrading.
!    Client version: 2.1
!    Server version: 2.0
username: admin
password: 
!    WARNING: Client and server API versions do not match. Please consider upgrading.
!    Client version: 2.1
!    Server version: 2.0
Logged in as admin
$ cat ~/.deis/client.json 
{"username":"xxxx","ssl_verify":false,"controller":"<a v2.0.0 workflow endpoint>","token":"","response_limit":0}
```

I am not having this problem authenticating against `v2.1.0` workflow clusters when using v`2.1` of the client.

**Comment by bacongobbler** [comment]
I think this is exactly the reason why we decided to put that big fat warning in there. The client is not guaranteed to work with older releases, and vise versa. That being said, I don't see any reason why this would be broken according to https://github.com/deis/controller/compare/v2.0.0...v2.1.0. The only relevant change I've seen server-side is in https://github.com/deis/controller/commit/46829d34b99edea661075301c847cced26318da8 and that was in 2.0.0-alpha, so I'm at a loss how a 2.1 client would fail to save the token from a 2.0 server. I'll see if I can replicate :)

**Comment by Joshua-Anderson** [comment]
Ok. I figured out the problem.... I screwed up 😦. When the [request returns a version mismatch](https://github.com/deis/controller-sdk-go/blob/master/http.go#L68), it returns that error in addation to the response. When the [methods get the error response](https://github.com/deis/controller-sdk-go/blob/master/auth/auth.go#L41) they immediately bail out and don't attempt to parse the body.

The CLI on the other hand, checks to see if the error is an API mismatch, and if there is, recognizes it is a warning and continues on, even though they're only working with a blank object.

What the SDK should do is implement a method similar to the [workflow-cli's checkAPICompatibility](https://github.com/deis/workflow-cli/blob/master/cmd/utils.go#L78) which checks if the the error is an api compatibility one before bailing out.
